alacritty/src/macos/proc.rs

```diff
@@ -1,41 +1,84 @@
-use std::ffi::{CStr, CString};
+use std::ffi::{CStr, CString, IntoStringError};
+use std::fmt::{self, Display, Formatter};
 use std::io;
 use std::mem::{self, MaybeUninit};
-use std::os::raw;
+use std::os::raw::{c_int, c_void};
 use std::path::PathBuf;
 
-pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+/// Error occuring during getting cwd from `proc_pidinfo`.
+#[derive(Debug)]
+pub enum Error {
+    /// IO error using `proc_pidinfo`.
+    Io(io::Error),
+
+    /// Error converting into utf8 string.
+    IntoString(IntoStringError),
+
+    /// Invalid size.
```

Comment:
If we add a comment, it should be better at describing the actual problem:

```suggestion
    /// Expected struct size did not match libproc's.
```

The use of the word `struct` could be contested, just using `return size` would be fine too I think.